analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -63,16 +70,131 @@ public sealed class MemberInitializerRedundant : SonarDiagnosticAnalyzer
                     // interfaces cannot have instance fields and instance properties cannot have initializers
                     if (declaration is ClassDeclarationSyntax)
                     {
-                        InitializationVerifierFactory.InstanceMemberInitializationVerifier.Report(c, declaration, members);
+                        InitializationVerifierFactory.InstanceMemberInitializationVerifier.Report(c, declaration, members, forceSonarCfg);
                     }
-                    InitializationVerifierFactory.StaticMemberInitializationVerifier.Report(c, declaration, members);
+                    InitializationVerifierFactory.StaticMemberInitializationVerifier.Report(c, declaration, members, forceSonarCfg);
                 },
                 // For record support, see details in https://github.com/SonarSource/sonar-dotnet/pull/4756
                 // it is difficult to work with instance record constructors w/o raising FPs
                 SyntaxKind.ClassDeclaration,
                 SyntaxKind.StructDeclaration,
                 SyntaxKind.InterfaceDeclaration);
 
+        private class RedundancyChecker
```

Comment:
There's a lot of syntax stuff here that might not be needed at all in Operation-based implementation. Then it could be moved into the SonarCfg-specific part.